pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -905,6 +905,32 @@ final class Event {
           },
         );
 
+  /// Event that is sent from the Dart MCP server when a tool is invoked.
+  ///
+  /// The [client] is the name of the client, as given when it connected to the
+  /// MCP server, and [clientVersion] is the version of the client.
+  ///
+  /// The [serverVersion] is the version of the Dart MCP server.
+  ///
+  /// The [type] identifies the kind of event this is, and [eventData] is the
+  /// actual data for that event.
+  Event.dartMCPEvent({
+    required String client,
```

Comment:
Do we know that we'll only want to send analytics events when a client is connected? If there's any possibility we want to send analytics unrelated to a particular client, we should make the client parameters optional.

Reply:
The Dart MCP server is always started by a client so we will always have one.

Reply:
No I don't think so, everything is always in the context of a single client.